cpp/ql/lib/semmle/code/cpp/dataflow/internal/FlowSummaryImpl.qll

```diff
@@ -0,0 +1,248 @@
+    )
+  }
+
+  /**
+   * Holds if an external sink specification exists for `e` with input specification
+   * `input`, kind `kind` and provenance `provenance`.
+   */
+  predicate sinkElement(SourceOrSinkElement e, string input, string kind) {
+    exists(
+      string package, string type, boolean subtypes, string name, string signature, string ext
+    |
+      sinkModel(package, type, subtypes, name, signature, ext, input, kind, _) and
+      e = interpretElement(package, type, subtypes, name, signature, ext)
+    )
+  }
+
+  private newtype TInterpretNode =
+    TElement_(Element n) or
+    TNode_(Node n) or
+    TDataFlowCall_(DataFlowCall c)
+
+  /** An entity used to interpret a source/sink specification. */
+  class InterpretNode extends TInterpretNode {
+    /** Gets the element that this node corresponds to, if any. */
+    SourceOrSinkElement asElement() { this = TElement_(result) }
+
+    /** Gets the data-flow node that this node corresponds to, if any. */
+    Node asNode() { this = TNode_(result) }
+
+    /** Gets the call that this node corresponds to, if any. */
+    DataFlowCall asCall() { this = TDataFlowCall_(result) }
+
+    /** Gets the callable that this node corresponds to, if any. */
+    DataFlowCallable asCallable() { result.(Function) = this.asElement() }
+
+    /** Gets the target of this call, if any. */
+    Element getCallTarget() {
+      result = this.asNode().asExpr().(Call).getTarget()
+    }
+
+    /** Gets a textual representation of this node. */
+    string toString() {
+      result = this.asElement().toString()
+      or
+      result = this.asNode().toString()
+      or
+      result = this.asCall().toString()
+    }
+
+    /** Gets the location of this node. */
+    Location getLocation() {
+      result = this.asElement().getLocation()
+      or
+      result = this.asNode().getLocation()
+      or
+      result = this.asCall().getLocation()
+    }
+  }
+
+  /** Provides additional sink specification logic. */
+  bindingset[c]
+  predicate interpretOutput(string c, InterpretNode mid, InterpretNode node) {
+    // Allow fields to be picked as output nodes.
+    exists(Node n, Element ast |
+      n = node.asNode() and
+      ast = mid.asElement()
+    |
+      c = "" and
+      n.asExpr().(VariableAccess).getTarget() = ast
+    )
+  }
+
+  /** Provides additional source specification logic. */
+  bindingset[c]
+  predicate interpretInput(string c, InterpretNode mid, InterpretNode node) {
+    exists(Node n, Element ast, VariableAccess e |
+      n = node.asNode() and
+      ast = mid.asElement() and
+      e.getTarget() = ast
+    |
+      // Allow fields to be picked as input nodes.
+      c = "" and
+      e.getQualifier() = n.asExpr()
+      or
+      // Allow post update nodes to be picked as input nodes when the `input` column
+      // of the row is `PostUpdate`.
+      c = "PostUpdate" and
+      e.getQualifier() = n.(PostUpdateNode).getPreUpdateNode().asExpr()
+    )
```

Comment:
Similarly, I don't think we should include anything here until we have a clear use-case for it.

Reply:
The counter-argument is that it's frustrating when you're writing CSV models to find that basic features aren't supported, so you have to pause your work and go and implement them.

That said, I should ensure this case has test coverage if I'm to keep it.

Reply:
... it doesn't have test coverage.  I've added a `TODO` comment to make sure I come back to this.